packages/block-library/src/heading/transforms.js

```diff
@@ -67,6 +67,16 @@ const transforms = {
 				} );
 			},
 		} ) ),
+		...[ 1, 2, 3, 4, 5, 6 ].map( ( level ) => ( {
+			type: 'enter',
```

Comment:
I didn't know about these kinds of transforms. I wonder if there's a way to make them more obvious in the UI but this works for now.

Should we make it case insensitive?

Reply:
> Should we make it case insensitive?

Good point, they should work both for `/h3` and `/H3`. Pushed an additional commit on https://github.com/WordPress/gutenberg/pull/26597/commits/dcf3909f3714b3a233f05cd24696bad5ec61c471 👍